src/aspire/basis/steerable.py

```diff
@@ -319,3 +338,153 @@ def blk_diag_cov_shape(self):
+        angular_mask = signs_mask.copy()
+
+        if radial is None:
+            radial_mask[:] = True
+        else:
+            for k in np.atleast_1d(radial):
+                radial_mask[self.radial_indices == k] = True
+
+        if angular is None:
+            angular_mask[:] = True
+        else:
+            for el in np.atleast_1d(angular):
+                angular_mask[self.angular_indices == el] = True
+
+        if signs is None:
+            signs_mask[:] = True
+        else:
+            for s in np.atleast_1d(signs):
+                signs_mask[self.signs_indices == s] = True
+
+        mask = radial_mask & angular_mask & signs_mask
+
+        return mask
+
+    def to_real(self, complex_coef):
+        """
+        Return real valued representation of complex coefficients.
+        This can be useful when comparing, prototyping, or
+        implementing methods from literature.
+
+        There is a corresponding method, `to_complex`.
+
+        :param complex_coef: Complex `Coef` from this basis.
+        :return: Real `Ceof` representation from this basis.
+        """
+
+        if not isinstance(complex_coef, Coef):
+            raise TypeError(
+                f"complex_coef should be instance of `Coef`, received {type(complex_coef)}."
+            )
+
+        if complex_coef.dtype not in (np.complex128, np.complex64):
+            raise TypeError("coef provided to to_real should be complex.")
+
+        # Pass through dtype precisions, but check and warn if mismatched.
+        dtype = real_type(complex_coef.dtype)
+        if dtype != self.dtype:
+            logger.warning(
+                f"Complex coef dtype {complex_coef.dtype} does not match precision of basis.dtype {self.dtype}, returning {dtype}."
+            )
+
+        coef = np.zeros((*complex_coef.stack_shape, self.count), dtype=dtype)
+        complex_coef = complex_coef.asnumpy()
+
+        ind = 0
+        idx = np.arange(self.k_max[0], dtype=int)
+        ind += np.size(idx)
+        ind_pos = ind
+
+        coef[..., idx] = complex_coef[..., idx].real
+
+        for ell in range(1, self.ell_max + 1):
+            idx = ind + np.arange(self.k_max[ell], dtype=int)
+            idx_pos = ind_pos + np.arange(self.k_max[ell], dtype=int)
+            idx_neg = idx_pos + self.k_max[ell]
+
+            c = complex_coef[..., idx]
+            coef[..., idx_pos] = 2.0 * np.real(c)
+            coef[..., idx_neg] = -2.0 * np.imag(c)
+
+            ind += np.size(idx)
+            ind_pos += 2 * self.k_max[ell]
+
+        return Coef(self, coef)
+
+    def to_complex(self, coef):
+        """
+        Return complex valued representation of complex coefficients.
+        This can be useful when comparing, prototyping, or
+        implementing methods from literature.
+
+        There is a corresponding method, `to_real`.
+
+        :param coef: Real `Coef` from this basis.
+        :return: Complex `Coef` representation from this basis.
```

Comment:
We've talked about something like `ComplexCoef` before, but I really didn't think it would add much value after beginning the push of the steerable 2D basis classes towards real coeffs.  It should not be a lot of code to implement, mainly tests.  I'll break it up for the re-review.

Reply:
Was able to add this with only a few test changes.  Will push and look if anything important missing in the coverage and begin the re-reviews.

Reply:
Looks great. Thanks!